DependencyInjection/CompilerPass/PluginPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Http\HttplugBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Remove plugins if their dependencies are missing.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class PluginPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!($container->hasDefinition('logger') || $container->hasAlias('logger'))) {
+            $container->removeDefinition('httplug.plugin.logger');
+        }
+
+        if (!$container->hasDefinition('debug.stopwatch')) {
+            $container->removeDefinition('httplug.plugin.stopwatch');
+        }
+    }
```

Comment:
That would require us to know what plugins that are being used. We have currently no clue about that in the compiler pass. It could be solved with container parameters or a DebugPluginClient::getPlugins(). 

You could also make an argument that the very few users that have logger/stopwatch disabled probably knows about it and are skilled enough to figure out that the LoggerPlugin requires a logger. But now I am making assumptions based on no facts.